app/src/main/java/org/sopt/study/mvvmstudy/MainActivity.kt

```diff
@@ -0,0 +1,58 @@
+package org.sopt.study.mvvmstudy
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import org.sopt.study.mvvmstudy.data.Contact
+import org.sopt.study.mvvmstudy.data.ContactDatabase
+import org.sopt.study.mvvmstudy.data.ContactViewModel
+import org.sopt.study.mvvmstudy.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var contactViewModel: ContactViewModel
+    private lateinit var binding : ActivityMainBinding
+    private lateinit var contactAdapter : ContactAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+
+        initAdapter()
+
+        contactViewModel = ViewModelProviders.of(this).get(ContactViewModel::class.java)
+        contactViewModel.getAll().observe(this, Observer<List<Contact>>{contacts->contactAdapter.setContacts(contacts!!)})
+
+        binding.btnAdd.setOnClickListener{
+            val intent = Intent(this, AddActivity::class.java)
+            startActivity(intent)
+        }
+
+        setContentView(binding.root)
+    }
+
+    private fun initAdapter(){
+        contactAdapter = ContactAdapter({contact->
+            val intent = Intent(this, AddActivity::class.java)
+            intent.putExtra(AddActivity.EXTRA_CONTACT_NAME, contact.name)
+            intent.putExtra(AddActivity.EXTRA_CONTACT_NUMBER, contact.number)
+            intent.putExtra(AddActivity.EXTRA_CONTACT_ID, contact.id)
```

Comment:
이 부분은 스코프 함수를 활용해서
```suggestion
            val intent = Intent(this, AddActivity::class.java).apply {
                       putExtra(AddActivity.EXTRA_CONTACT_NAME, contact.name)
                       putExtra(AddActivity.EXTRA_CONTACT_NUMBER, contact.number)
                       putExtra(AddActivity.EXTRA_CONTACT_ID, contact.id)
            }
```
요런식으로 좀 더 간결하게 표현할 수도 있을 것 같네용~^_^ 스코프 함수 종류는 많으니까 더 찾아보시기 바랍니당~!!